widgetry/src/event.rs

```diff
@@ -6,13 +7,32 @@ use geom::Duration;
 
 use crate::{EventCtx, Line, ScreenDims, ScreenPt, TextSpan};
 
+// Mouse-up events longer than this will be considered single clicks
+// Ideally the delay would be a little more tolerant - e.g. 500m, but because we don't actually
+// have a way to indicate that a single click was handled (and thus *shouldn't* be counted as part of a double click)
+// it's too easy to have false positives.
+const MAX_DOUBLE_CLICK_DURATION: instant::Duration = instant::Duration::from_millis(300);
+
 #[derive(Debug, Clone, Copy, PartialEq)]
 pub enum Event {
     // Used to initialize the application and also to recalculate menu state when some other event
     // is used.
     NoOp,
     LeftMouseButtonDown,
-    LeftMouseButtonUp,
+    /// Note: When double clicking, there will be two `LeftMouseButtonUp` events in short
+    /// succession - first a `LeftMouseButtonUp { is_double_click: false }`, followed by
+    /// a `LeftMouseButtonUp { is_double_click: true }`.
+    ///
+    /// This was done for ease of implementation - it allows a target to ignore single clicks and
```

Comment:
I think this is quite a clever implementation and indeed avoids boiling the ocean. Since we don't have any anticipated needs for both in one place, let's go for it!